### PR TITLE
fix: make getMountedControllers support nested controllers

### LINKED ISSUE #1802: [BUG] Missing nested controllers when using `pathPatterns` on swagger settings
## Information

- **Version:** 6.105.2
- Packages:

Missing nested controllers when using `pathPatterns` on swagger settings.

## Example

```ts
import "@tsed/ajv";
import { PlatformApplication } from "@tsed/common";
import { Configuration, Controller, Inject } from "@tsed/di";
import "@tsed/platform-express"; // /!\ keep this import
import { Get } from "@tsed/schema";
import "@tsed/swagger";

@Controller("/articles")
export class ArticlesController {
  @Get("/")
  get() {
    return "articles";
  }
}

@Controller({
  path: "/home",
  children: [ArticlesController]
})
export class HomeController {
  @Get("/")
  get() {
    return "home";
  }
}

@Controller({
  path: "/api",
  children: [HomeController],
})
export class ApiController {
  @Get("/")
  get() {
    return "api";
  }
}

@Configuration({
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8086,
  httpsPort: false, // CHANGE
  componentsScan: false,
  mount: {
    "/": [ApiController],
  },
  swagger: [
    {
      path: "/doc",
      pathPatterns: ["/api/home/**"],
      specVersion: "3.0.1",
    },
  ],
})
export class Server {
  @Inject()
  protected app: PlatformApplication;

  @Configuration()
  protected settings: Configuration;
}
```

## Acceptance criteria

- [ ] It should be possible to extract the nested controllers and display on swagger.json

### PR DESCRIPTION
**Pull request by weirongxu**
## Information

| Type                  | Breaking change |
| --------------------- | --------------- |
| Fix | No          |

---

Fixes #1802

## Todos

- [x] Tests
- [x] Coverage
- [ ] Example
- [ ] Documentation

### CODE REVIEW

File: packages/platform/common/src/services/Platform.ts [hunk 1]
```diff
@@ -46,12 +47,10 @@ export class Platform {
       return;
     }
 
-    const ctrlPath = concatPath(basePath, JsonEntityStore.from(provider.token).path);
+    this._controllers.push(...this.getAllControllers(basePath, token));
+    console.dir(this.getAllControllers(basePath, token));
```
Comment (Romakita) [inline packages/platform/common/src/services/Platform.ts]: console.log to be removed :)

**Review by Romakita — CHANGES_REQUESTED** [review]

File: packages/platform/common/src/services/Platform.ts [hunk 2]
```diff
@@ -103,6 +106,28 @@ export class Platform {
     });
   }
 
+  /**
+   * Get all router controllers from the controller token.
+   * @private
+   */
+  private getAllControllers(basePath: string, token: Type<any> | any): RouteController[] {
```
Comment (Romakita) [inline packages/platform/common/src/services/Platform.ts:112]: I'm not sure that is the right place to apply the change. 
The SwaggerService build recursively the routes:
https://github.com/tsedio/tsed/blob/cd4310d2e4b3e9786916afb3287e20497ed6385c/packages/specs/swagger/src/services/SwaggerService.ts

```typescript
  protected buildRoutes(ctrl: ControllerProvider, options: SpecSerializerOptions) {
    const rootPath = options.rootPath + ctrl.path;

    ctrl.children
      .map((ctrl) => this.injectorService.getProvider(ctrl))
      .forEach((provider: ControllerProvider) => {
        if (!provider.store.get("hidden")) {
          const spec = this.buildRoutes(provider, {
            ...options,
            rootPath
          });

          options.append(spec);
        }
      });

    return getSpec(ctrl.token, options);
  }
```

I think there is propably a conflict between your change and the current behavior the SwaggerService...

Comment (Romakita) [reply packages/platform/common/src/services/Platform.ts:112]: In the builRoutes, I think we can remove the following code:

```
ctrl.children
      .map((ctrl) => this.injectorService.getProvider(ctrl))
      .forEach((provider: ControllerProvider) => {
        if (!provider.store.get("hidden")) {
          const spec = this.buildRoutes(provider, {
            ...options,
            rootPath
          });

          options.append(spec);
        }
      });

```

Comment (Romakita) [reply packages/platform/common/src/services/Platform.ts:112]: @weirongxu

Comment (weirongxu) [reply packages/platform/common/src/services/Platform.ts:112]: @Romakita, Got it, but there are some unit tests to adjust after removing the code

**Review by Romakita — APPROVED** [review]

**Pull request merged**